main.py

```diff
@@ -725,24 +802,46 @@ def _send_flow_mods(
         flows,
         reraise_conn=True,
         send_barrier=ENABLE_BARRIER_REQUEST,
+        batch_size=0,
+        batch_interval=0,
     ):
         """Send FlowMod (and BarrierRequest) given a list of flow_dicts to switches."""
+        flows_zipped = list((fmod, flow) for fmod, flow in zip(flow_mods, flows))
+        batch_size = (
+            min(batch_size, len(flows_zipped))
+            if batch_size > 0
+            else max(1, len(flows_zipped))
+        )
+        batch_interval = max(0, batch_interval)
         for switch in switches:
-            for i, (flow_mod, flow) in enumerate(zip(flow_mods, flows)):
-                try:
-                    self._send_flow_mod(switch, flow_mod)
-                    if send_barrier and i == len(flow_mods) - 1:
-                        self._send_barrier_request(switch, flow_mods)
-                except SwitchNotConnectedError:
-                    if reraise_conn:
-                        raise
-                with self._flow_mods_sent_lock:
-                    self._add_flow_mod_sent(
-                        flow_mod.header.xid,
-                        flow,
-                        build_command_from_flow_mod(flow_mod),
+            try:
+                k = 0
+                for i in range(0, len(flows_zipped), batch_size):
```

Comment:
I'll push a new unit test covering the slicing later on, but I'll go ahead and maintain as ready to review.

Reply:
Covered here 44e52d1186406f774b0bca103349c222a3348bd0